toolchain/check/import_ref.cpp

```diff
@@ -258,13 +284,47 @@ class ImportRefResolver {
     return import_name_id;
   }
 
+  // Translates a NameScopeId from the import IR to a local NameScopeId. Adds
+  // unresolved constants to the work stack.
+  auto GetLocalNameScopeId(SemIR::NameScopeId name_scope_id)
+      -> SemIR::NameScopeId {
+    if (!name_scope_id.is_valid()) {
+      return SemIR::NameScopeId::Invalid;
+    }
+    auto inst_id = import_ir_.name_scopes().Get(name_scope_id).inst_id;
+    if (!inst_id.is_valid()) {
+      // Map scopes that aren't associated with an instruction to invalid
+      // scopes. For now, such scopes aren't used, and we don't have a good way
+      // to rmmap them.
+      return SemIR::NameScopeId::Invalid;
+    }
+    auto const_id = GetLocalConstantId(inst_id);
+    if (!const_id.is_valid()) {
+      return SemIR::NameScopeId::Invalid;
+    }
+    auto name_scope_inst = context_.insts().Get(const_id.inst_id());
+    if (auto namespace_inst = name_scope_inst.TryAs<SemIR::Namespace>()) {
+      return namespace_inst->name_scope_id;
+    }
+    if (auto class_inst = name_scope_inst.TryAs<SemIR::ClassType>()) {
+      return context_.classes().Get(class_inst->class_id).scope_id;
+    }
+    if (auto interface_inst = name_scope_inst.TryAs<SemIR::InterfaceType>()) {
+      return context_.interfaces().Get(interface_inst->interface_id).scope_id;
+    }
+    if (const_id == SemIR::ConstantId::Error) {
+      return SemIR::NameScopeId::Invalid;
+    }
```

Comment:
Why is this validation after const_id.inst_id() is used? Wouldn't using the result of insts().Get be an issue for an Error?

Reply:
`ConstantId::Error` is just the constant form of `InstId::BuiltinError`, so it's a `Builtin` instruction, and `insts().Get` on it is fine. Given that this is a rare error case, I put it here so we don't need to check it on the common case.